src/chart/generateCategoricalChart.tsx

```diff
@@ -1198,48 +1198,13 @@ const generateCategoricalChart = ({
       }
     };
 
-    handleReceiveSyncEvent = (cId: any, chartId: any, data: any) => {
-      const { syncId, layout } = this.props;
-      const { updateId } = this.state;
+    handleReceiveSyncEvent = (cId: number | string, chartId: string, data: CategoricalChartState) => {
+      const { syncId } = this.props;
 
       if (syncId === cId && chartId !== this.uniqueChartId) {
-        const { dataStartIndex, dataEndIndex } = data;
-
-        if (!_.isNil(data.dataStartIndex) || !_.isNil(data.dataEndIndex)) {
-          this.setState({
-            dataStartIndex,
-            dataEndIndex,
-            ...this.updateStateOfAxisMapsOffsetAndStackGroups({
-              props: this.props,
-              dataStartIndex,
-              dataEndIndex,
-              updateId,
-            }),
-          });
-        } else if (!_.isNil(data.activeTooltipIndex)) {
-          const { chartX, chartY, activeTooltipIndex } = data;
-          const { offset, tooltipTicks } = this.state;
-          if (!offset) {
-            return;
-          }
-          const viewBox: ViewBox = { ...offset, x: offset.left, y: offset.top };
-          // When a categotical chart is combined with another chart, the value of chartX
-          // and chartY may beyond the boundaries.
-          const validateChartX = Math.min(chartX, viewBox.x + viewBox.width);
-          const validateChartY = Math.min(chartY, viewBox.y + viewBox.height);
-          const activeLabel = tooltipTicks[activeTooltipIndex] && tooltipTicks[activeTooltipIndex].value;
-          const activePayload: any = this.getTooltipContent(activeTooltipIndex);
-          const activeCoordinate = tooltipTicks[activeTooltipIndex]
-            ? {
-                x: layout === 'horizontal' ? tooltipTicks[activeTooltipIndex].coordinate : validateChartX,
-                y: layout === 'horizontal' ? validateChartY : tooltipTicks[activeTooltipIndex].coordinate,
-              }
-            : originCoordinate;
-
-          this.setState({ ...data, activeLabel, activeCoordinate, activePayload });
-        } else {
-          this.setState(data);
-        }
+        (window.requestAnimationFrame || window.setImmediate || window.setTimeout)(
```

Comment:
this line may  throw error in node?

Reply:
i'm guessing that would be in SSR? and so, the environment for SSR doesn't provide a `window` global?